File: src/main/Java/boj_step15/boj1934.java
import java.io.*;
import java.util.StringTokenizer;

public class boj1934 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int t = Integer.parseInt(reader.readLine());
        while (t-- > 0) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            int a = Integer.parseInt(tokenizer.nextToken());
            int b = Integer.parseInt(tokenizer.nextToken());
            writer.append((a * b) / gcd(a, b) + "\n");
        }
        writer.close();
    }
    private static int gcd(int a, int b) { if (b != 0) return gcd (b, a % b); else return a; }
}

File: src/main/boj_step15/boj1934.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

class boj1934 {
    fun main() {
        val r = BufferedReader(InputStreamReader(System.`in`))
        val w = BufferedWriter(OutputStreamWriter(System.out))
        val n = r.readLine().toInt()
        repeat(n) {
            val (a, b) = r.readLine().split(" ").map { it.toInt() }
            w.write(((a * b) / gcd(a, b)).toString())
            w.newLine()
        }
        w.close()
    }

    private fun gcd(a: Int, b: Int): Int = if(b != 0) gcd(b, a % b) else a
}